src/test/java/ai/promoted/delivery/client/PromotedDeliveryClientTest.java

```diff
@@ -253,74 +253,74 @@ void testHasControlCohortMembershipCallsSDKDeliveryAndLogs() throws Exception {
+    assertEquals(10, resp.getResponse().getInsertionCount());
     verify(apiFactory.getSdkDelivery(), times(1)).runDelivery(dreq);
     verifyNoInteractions(apiFactory.getApiDelivery());
 
     ArgumentCaptor<LogRequest> logRequestCaptor = ArgumentCaptor.forClass(LogRequest.class);
     verify(apiFactory.getApiMetrics()).runMetricsLogging(logRequestCaptor.capture());
     LogRequest logRequest = logRequestCaptor.getValue();
 
-    assertSDKLogRequest(req, resp.getResponse(), logRequest);
+    assertSDKLogRequest(reqBuilder, resp.getResponse(), logRequest);
     assertDeliveryResponse(resp, ExecutionServer.SDK);
   }
 
   @Test
   void testApiDeliveryErrorFallsBackToSdkDelivery() throws Exception {
     PromotedDeliveryClient client = createDefaultClient();
 
-    Request req = new Request().insertion(TestUtils.createTestRequestInsertions(10));
-    DeliveryRequest dreq = new DeliveryRequest(req, null, false, 0);
+    Request.Builder reqBuilder = Request.newBuilder().addAllInsertion(TestUtils.createTestRequestInsertions(10));
+    DeliveryRequest dreq = new DeliveryRequest(reqBuilder, null, false, 0);
 
 
     when(apiFactory.getApiDelivery().runDelivery(any())).thenThrow(DeliveryException.class);
     when(apiFactory.getSdkDelivery().runDelivery(any()))
-        .thenReturn(new Response().insertion(req.getInsertion()));
+        .thenReturn(Response.newBuilder().addAllInsertion(reqBuilder.getInsertionList()).build());
 
     DeliveryResponse resp = client.deliver(dreq);
-    assertTrue(req.getClientRequestId().length() > 0);
+    assertTrue(reqBuilder.getClientRequestId().length() > 0);
 
-    assertEquals(10, resp.getResponse().getInsertion().size());
+    assertEquals(10, resp.getResponse().getInsertionCount());
     verify(apiFactory.getApiDelivery(), times(1)).runDelivery(dreq);
     verify(apiFactory.getSdkDelivery(), times(1)).runDelivery(dreq);
 
     ArgumentCaptor<LogRequest> logRequestCaptor = ArgumentCaptor.forClass(LogRequest.class);
     verify(apiFactory.getApiMetrics()).runMetricsLogging(logRequestCaptor.capture());
     LogRequest logRequest = logRequestCaptor.getValue();
 
-    assertSDKLogRequest(req, resp.getResponse(), logRequest);
+    assertSDKLogRequest(reqBuilder, resp.getResponse(), logRequest);
     assertDeliveryResponse(resp, ExecutionServer.SDK);
   }
 
   @Test
   void testPassThroughSdkInsertionIds() throws Exception {
     PromotedDeliveryClient client = createDefaultClient();
 
-    Request req = new Request().insertion(TestUtils.createTestRequestInsertions(10));
-    List<Insertion> insertions = req.getInsertion();
-    for (int i = 0; i < insertions.size(); i++) {
-      insertions.set(i, insertions.get(i).toBuilder().setInsertionId("ins" + i).build());
+    Request.Builder reqBuilder = Request.newBuilder().addAllInsertion(TestUtils.createTestRequestInsertions(10));
+    List<Insertion.Builder> insBuilders = reqBuilder.getInsertionBuilderList();
+    for (int i = 0; i < insBuilders.size(); i++) {
```

Comment:
Nit - I'd guess you don't need the extra list.  You can probably use getInsertionBuilder(index).

Reply:
Done